app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,72 @@
+class CartsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_common_data, only: [:show, :checkout, :index]
+
+  def show
+    @total_price = @cart.line_items.sum { |line_item| line_item.product.total_price * line_item.quantity }
+    
+    if request.post?
+      selected_address = Address.find(params[:address_id])
+      order = current_user.orders.create(address: selected_address, total_price: @total_price)
+      
+      @cart.line_items.each do |line_item|
+        order.order_items.create(
+          product: line_item.product,
+          quantity: line_item.quantity,
+          price: line_item.product.total_price
+        )
+      end
+      
+      @cart.destroy
```

Comment:
Why do you destroy the cart after showing it ?? :thinking:

Reply:
I think the whole show method is wrong

Reply:
We have already deleted the entire method. We had it too much, because it does not affect the operation of the page and did the same as the checkout.